### DISCUSSION TITLE
Conribute unused selfhosted runners

### DISCUSSION THREAD

**Discussion by Pavlinchen** [community/community#15213] [Actions] [answered]
Hi,
I'd personally love to contribute a few compute ressources to the GitHub provided runners for general use.
Maybe that's not doable due to security concerns or something (would it help to limit it to public repos, or provide a check mark at the users own risk)?
So my Idea was that you can either 
1. Enable a flag on an existing selfhosted runner to make it available to the public, when it's not used, but cut the queue (so be the next job in line or maybe even start parallel (after a time)), if it's supposed to do something. or
2. Have a dedicated docker/VM/executeable or similar (maybe even locked up, so the host can't tamper with it), that can be run on spare hardware, for people who don't need the runner themselves, but want to contribute to the awesomeness, that GitHub is, without the hastle, of digging into the setup process.

I don't know, how doable that is, but I like the idea.
Any feedback is appreciated

**Comment by ethomson** [comment, marked as answer]
That's a very generous offer, @pavlinchen. It would be difficult for the reason you mention - security would be very difficult, in both directions. A generous person like yourself, hosting a computing resource, could have it tampered with by a less trustworthy user. Even running within a container [can have escapes](https://blog.trailofbits.com/2019/07/19/understanding-docker-container-escapes/) where somebody could end up running a malicious workflow and then able to modify your computing resource directly.

Virtual machines have much tighter security boundaries - but it would be hard for us to verify that somebody wasn't bringing a malicious virtual machine to the party. Since workflows get limited-use GitHub tokens, those are sent to the runner, so we'd need to be very careful about who was able to bring runners and that we could inspect them somehow to ensure that they were not tampered with.

Anyway, it's a _very_ cool idea, but I regret that it's not easy or obvious how to implement it. Something that we might want to think about in the future, though!

**Reply by Pavlinchen** [reply]
Hi,
thanks for taking the time to answer.
Yeah so unfortunately I managed to figure that much :/

Welp anyways thanks for the reply and have a great day! :)
